### ISSUE TITLE
ANCM: Inprocess: forwardWindowsAuthToken does not work

### ISSUE THREAD

**Issue by jhkimnew** [aspnet/IISIntegration#538]
1. Enable NTLM authentication for an aspnetcore app which is running in inprocess
2. Send a request to the aspnetcore app

EXPECTED:

The request headers should contain the following headers:

MS-ASPNETCORE-TOKEN
MS-ASPNETCORE-WINAUTHTOKEN

ACTUAL:

The request headers does not have the expected request headers.

**Comment by Tratcher** [comment]
No, this flag has no function for the in-process scenario. The MS- auth headers are not required for in-proc, we can query the IIS data structures directly.

**Comment by Tratcher** [comment]
Note the schema default was changed to `true` some time ago. The more interesting question here is if we defined some suppression behavior for `false`.

**Comment by jhkimnew** [comment]
@Shirhatti, @pan-wang  I think we should ignore forwardWindowsAuthToken for inprocess mode. 
Would you confirm? The current behavior is that we don't do anything when the forwardWindowsAuthToken is false in inprocess.

**Comment by jkotalik** [comment]
We don't need to forward the windows auth token at all because we can always access it in inprocess. Maybe if we are running in process, we can log that the windows auth token doesn't do anything. Otherwise I would say this is a non issue.

**Comment by shirhatti** [comment]
Shouldn't the right behavior here be to suppress the windows auth token if `forwardWindowsAuthToken` is explicitly set to false. My motivation is to preserve as much parity in behavior between in-process and out-of-process.
If we don't suppress an app that works perfectly with in-process will break when you switch to out-of-process if it relied on this idiosyncratic behavior.

**Comment by Tratcher** [comment]
The equivalent would be suppressing HttpContext.User, which could be possible.

**Comment by shirhatti** [comment]
@Tratcher What should we do here? I don't feel strongly enough about it one way orthe other.

Option 1: Do nothing. You cannot suppress HttpContext.User
Option 2:  Suppress HttpContext.User based on your web.config

**Comment by Tratcher** [comment]
Brock or Dominick wanted the ability to suppress HttpContext.User as it messes you up if you're trying to do some other kind of auth for some endpoints. Today they do it with IISOptions.AutomaticAuthentication = false, and we should preserve an in-code equivalent.

I'd only preserve forwardWindowsAuthToken as an effort to make in-proc and out-of-proc consistent and compatible. I.e. for users already using this setting. This flag's not a high priority, I haven't seen it widely used, but one of the major goals for in-proc is a seamless transition from out-of-proc.
